source/PlayerInfo.cpp

```diff
@@ -1945,6 +1945,20 @@ void PlayerInfo::AddSpecialLog(const string &type, const string &name, const str
 
 
 
+void PlayerInfo::RemoveSpecialLog(const std::string &type, const std::string &name)
+{
+	specialLogs[type].erase(specialLogs[type].find(name));
+}
+
+
+
+void PlayerInfo::RemoveSpecialLog(const std::string &type)
+{
+	specialLogs.erase(specialLogs.find(type));
```

Comment:
Similarly, you need to establish that `type` is a key in `specialLogs` before attempting to erase it.